src/Compilers/CSharp/Test/Symbol/Symbols/StaticAbstractMembersInInterfacesTests.cs

```diff
@@ -5764,6 +5764,55 @@ class Test
                 );
         }
 
+        [Fact]
+        public void ConsumeAbstractStaticMethod_06()
+        {
+            var source1 =
+@"
+public interface I1
+{
+    abstract static void M01();
+}
+";
+            var source2 =
+@"
+class Test
+{
+    static void M02<T>() where T : I1
+    {
+        T.M01();
+    }
+}
+";
+            var compilation1 = CreateCompilation(source1, options: TestOptions.DebugDll,
+                                                 parseOptions: TestOptions.RegularPreview,
+                                                 targetFramework: TargetFramework.NetCoreApp);
+
+            var compilation2 = CreateCompilation(source2, options: TestOptions.DebugDll,
+                                                 parseOptions: TestOptions.Regular9,
+                                                 targetFramework: TargetFramework.NetCoreApp,
+                                                 references: new[] { compilation1.ToMetadataReference() });
+
+            compilation2.VerifyDiagnostics(
+                // (6,9): error CS8652: The feature 'static abstract members in interfaces' is currently in Preview and *unsupported*. To use Preview features, use the 'preview' language version.
+                //         T.M01();
+                Diagnostic(ErrorCode.ERR_FeatureInPreview, "T").WithArguments("static abstract members in interfaces").WithLocation(6, 9)
+                );
+
+            var compilation3 = CreateCompilation(source2 + source1, options: TestOptions.DebugDll,
+                                                 parseOptions: TestOptions.Regular9,
+                                                 targetFramework: TargetFramework.NetCoreApp);
+
+            compilation3.VerifyDiagnostics(
+                // (6,9): error CS8652: The feature 'static abstract members in interfaces' is currently in Preview and *unsupported*. To use Preview features, use the 'preview' language version.
```

Comment:
It looks like the LangVersion diagnostic for the method call is given whether the method is from source or metadata. But for operators (like in `ConsumeAbstractUnaryOperator_06`), the LangVersion diagnostic is only given on the usage if the operator is from metadata. This seems inconsistent.

Reply:
>It looks like the LangVersion diagnostic for the method call is given whether the method is from source or metadata. But for operators (like in ConsumeAbstractUnaryOperator_06), the LangVersion diagnostic is only given on the usage if the operator is from metadata. This seems inconsistent.

There is no requarement to have this level of consistency. Reduction of diagnostics is desired when comes at a reasonable cost. When we are performing the check for operators we know what operator is used, when we performing the check for methods we don't know the method. It will be known after overload resolution.  

---
In reply to: [614431733](https://github.com/dotnet/roslyn/pull/52659#discussion_r614431733) [](ancestors = 614431733)

Reply:
I guess I could move the check into ```CheckReceiverAndRuntimeSupportForSymbolAccess``` helper, but there we don't really know if the receiver is coming from syntax and it feels like it is better to have the check close to the code that binds syntax.

---
In reply to: [614524424](https://github.com/dotnet/roslyn/pull/52659#discussion_r614524424) [](ancestors = 614524424,614431733)

Reply:
Thanks for the context. Don't worry about refactoring too much to accomplish this.